Library/ViewModels/RewardAddOnSelectionViewModel.swift

```diff
@@ -256,18 +256,79 @@ public final class RewardAddOnSelectionViewModel: RewardAddOnSelectionViewModelT
     .map(PledgeViewData.init)
     .takeWhen(self.continueButtonTappedProperty.signal)
 
+    let calculatedShippingTotal = Signal.combineLatest(
+      shippingRule.skipNil(),
+      allRewards,
+      selectedQuantities
+    )
+    .map(calculateShippingTotal)
+
+    let shippingTotal = Signal.zip(project, baseReward)
+      .map(getBaseRewardShippingTotal)
+
+    let allRewardsShippingTotal = Signal.merge(
+      shippingTotal,
+      calculatedShippingTotal
+    )
+
+    // Addtional pledge amount is zero if not backed.
+    let additionalPledgeAmount = Signal.merge(
+      configData.filter { $0.project.personalization.backing == nil }.mapConst(0.0),
+      project.map { $0.personalization.backing }.skipNil().map(\.bonusAmount)
+    )
+    .take(first: 1)
+
+    let allRewardsTotal = Signal.combineLatest(
+      selectedRewards,
+      selectedQuantities
+    )
+    .map(calculateAllRewardsTotal)
+
+    let calculatedPledgeTotal = Signal.combineLatest(
```

Comment:
This is an opinion but the variable name here is also not very descriptive. It should probably be something like `combinedPledgeTotal` or something along those lines.